chain/chain/src/state_sync/utils.rs

```diff
@@ -290,45 +276,44 @@ impl Chain {
         Ok(Some(sync_hash))
     }
 
-    /// Returns the list of extra block hashes for blocks that should be
-    /// downloaded before the state sync. The extra blocks are needed when there
-    /// are missing chunks in blocks leading to the sync hash block. We need to
-    /// ensure that for every shard we have at least one new chunk.
+    /// Returns the hashes of all extra blocks which must be downloaded for state sync.
+    /// Excludes the sync hash block and its prev block.
+    ///
+    /// For each shard, the node needs blocks going back to (and including)
+    /// the last new chunk before the sync hash block, to be used in:
```

Comment:
in `save_epoch_new_chunks` it says we need two new chunks per shard

Reply:
I am also curious about this and would like to understand it better. I just added some test coverage with missing chunks leading up to the last new chunk before the sync hash to see if I could uncover it (latest commit) but it doesn't seem to cause any problems.